File: Angular/src/app/shared/models/Quiz.ts
import {Question} from './Question';
import {status} from './status';



export class Quiz {
    id?: string;
    userEmail: string;
    title: string;
    description: string;
    questions: Question[] = [];
    isSelected: boolean;
    score: number;
    status: status;
    duration: number;
    maxAttempts: number;
    category: string;
    course: string;
    showSummary ? = false;
    finalScore ? = 0;
}
export class QuizSubmission {
    quizID: string;
    answers: { questionID: string, answer: string }[] = [];
}
export class QuizSubmissionResult {
    quizID: string;
    score: number;
    maxScore: number;
    passed: boolean;
    status: status;
    submittedAt: Date;
    Answers: { questionID: string, answer: string }[] = [];
}

File: Spring Boot/Courzelo/src/main/java/org/example/courzelo/models/Quiz.java
package org.example.courzelo.models;

import lombok.Data;
import org.example.courzelo.dto.QuestionDTO;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@Document(collection = "quizzes")
@Data
public class Quiz {
    @Id
    private String id;
    private String title;
    private String description;
    private List<Question> questions;
    private boolean isSelected;
    private double score;
    private String Status;
    private int duration; // in minutes
    private int maxAttempts;
    private String category;


    /*   id?: string;
    title: string;
    description: string;
    questions: Question[] = [];
    isSelected: boolean;
    score: number;
    Status: string;
    duration: number;
    maxAttempts: number;
    category: string;;*/
}
